src/app/workspace/resource/resource-properties/resource-properties.component.ts

```diff
@@ -0,0 +1,305 @@
+     */
+    @Input() showAllProps = false;
+
+    @Output() referredResourceClicked: EventEmitter<ReadLinkValue> = new EventEmitter<ReadLinkValue>();
+
+    @Output() referredResourceHovered: EventEmitter<ReadLinkValue> = new EventEmitter<ReadLinkValue>();
+
+    addButtonIsVisible: boolean; // used to toggle add value button
+    addValueFormIsVisible: boolean; // used to toggle add value form field
+    propID: string; // used in template to show only the add value form of the corresponding value
+
+    valueOperationEventSubscription: Subscription; // [] = []; // array of ValueOperationEvent subscriptions
+
+    constructor(
+        @Inject(DspApiConnectionToken) private _dspApiConnection: KnoraApiConnection,
+        private _valueOperationEventService: ValueOperationEventService,
+        private _valueService: ValueService
+    ) { }
+
+    ngOnInit() {
+        if (this.parentResource) {
+            // get user permissions
+            const allPermissions = PermissionUtil.allUserPermissions(
+                this.parentResource.userHasPermission as 'RV' | 'V' | 'M' | 'D' | 'CR'
+            );
+
+            // if user has modify permissions, set addButtonIsVisible to true so the user see's the add button
+            this.addButtonIsVisible = allPermissions.indexOf(PermissionUtil.Permissions.M) !== -1;
+        }
+
+        // listen for the AddValue event to be emitted and call hideAddValueForm()
+        // this.valueOperationEventSubscription = this._valueOperationEventService.on(Events.ValueAdded, () => this.hideAddValueForm());
+
+        // subscribe to the ValueOperationEventService and listen for an event to be emitted
+        this.valueOperationEventSubscription = this._valueOperationEventService.on(
+            Events.ValueAdded, (newValue: AddedEventValue) => {
+                this.addValueToResource(newValue.addedValue);
+                this.hideAddValueForm();
+            });
+
+        this.valueOperationEventSubscription = this._valueOperationEventService.on(
+            Events.ValueUpdated, (updatedValue: UpdatedEventValues) => {
+                this.updateValueInResource(updatedValue.currentValue, updatedValue.updatedValue);
+                this.hideAddValueForm();
+            }
+        );
+
+        this.valueOperationEventSubscription = this._valueOperationEventService.on(
+            Events.ValueDeleted, (deletedValue: DeletedEventValue) => this.deleteValueFromResource(deletedValue.deletedValue)
+        );
+    }
+
+    ngOnDestroy() {
+        // unsubscribe from the event bus when component is destroyed
+        if (this.valueOperationEventSubscription !== undefined) {
+            this.valueOperationEventSubscription.unsubscribe();
+        }
+        // unsubscribe from the ValueOperationEventService when component is destroyed
+        // if (this.valueOperationEventSubscriptions !== undefined) {
+        //     this.valueOperationEventSubscriptions.forEach(sub => sub.unsubscribe());
```

Comment:
You'll need this code instead of the code above if you switch to using an array of subscriptions

Reply:
Yes, I know. I can try again...

Reply:
resolved in 7e78424